src/msac.rs

```diff
@@ -301,38 +296,40 @@ unsafe fn dav1d_msac_decode_symbol_adapt_rust(
 
 unsafe extern "C" fn dav1d_msac_decode_symbol_adapt_c(
     s: *mut MsacContext,
-    cdf: *mut uint16_t,
+    cdf: *mut u16,
     n_symbols: size_t,
 ) -> libc::c_uint {
-    dav1d_msac_decode_symbol_adapt_rust(s, cdf, n_symbols)
+    // # Safety
+    //
+    // This is only called from [`dav1d_msac_decode_symbol_adapt16`],
+    // where there is an `assert!(n_symbols < cdf.len());`.
+    // Thus, `n_symbols + 1` is a valid length for the slice `cdf` came from.
```

Comment:
Can we also cleanup `dav1d_msac_decode_symbol_adapt16` at the same time so that we don't need this unsafe conversion? Or are you planning to do that in a separate PR?

Reply:
The unsafe conversion is necessary here since it roundtrips through asm.  Slices are unstable across FFI boundaries, so an `unsafe` block is required here one way or another.